### ISSUE TITLE
Setting colors by API does not work anymore

### ISSUE THREAD

**Issue by janrudolph** [RocketChat/Rocket.Chat#9628]
### Description:

An admin is able to setup most Rocket.Chat settings by the API, e.g. setting up the colors (Layout -> colors). In previous versions this worked pretty fine. Since 0.59 setting colors by the API does not work anymore. 

### Server Setup Information:

* Version of Rocket.Chat Server: 0.61.0
* Operating System: Linux
* Deployment Method(snap/docker/tar/etc): AWS
* Number of Running Instances: 2
* Node Version: v8.9.3

### Steps to Reproduce:

1. /api/v1/login with admin
2. Example: Set "theme-color-rc-color-primary-darkest" with value "#03625A" by API 
=> color is not applied 

### Expected behavior:

Color is set to the correct value and "color-editor" is set to "color"

### Actual behavior:

Value is inserted in correct field, but "color-editor" is set to "expression" (not color!). An admin has to set the "color-editor" to "color" manually.

<img width="298" alt="colors-not-set" src="https://user-images.githubusercontent.com/3596824/35922693-7fe47d00-0c1e-11e8-9722-96140bd3e1a2.png">

**Comment by vitormattos** [comment]
What you make to change color? In mongodb has another attribute to define the type of color.

**Comment by mrsimpson** [comment]
in rocketchat-api/server/v1/settings.js, `post()`:

```
		if (setting.type === 'color' && this.bodyParams && this.bodyParams.editor && this.bodyParams.value) {
			RocketChat.models.Settings.updateOptionsById(this.urlParams._id, {editor: this.bodyParams.editor});
			RocketChat.models.Settings.updateValueNotHiddenById(this.urlParams._id, this.bodyParams.value);
			return RocketChat.API.v1.success();
		}
```
There are other types of settings which could use special handling.
@graywolf336 Any objections solving this in the API-layer: https://github.com/assistify/Rocket.Chat/blob/d3f881d0938c3192f15d70d12571ec89ca542d6d/packages/rocketchat-api/server/v1/settings.js#L76

**Comment by graywolf336** [comment]
@mrsimpson my apologies, I totally missed your comment.

**Comment by rodrigok** [comment]
Closed via #10488
